clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -90,7 +90,7 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
       mUfsPath = fileInfo.getUfsPath();
       String fileName = PathUtils.temporaryFileName(fileId, mNonce, mUfsPath);
       UnderFileSystem ufs = UnderFileSystem.get(fileName, ClientContext.getConf());
-      String parentPath = (new TachyonURI(mUfsPath)).getParent().getPath();
+      String parentPath = (new TachyonURI(mUfsPath)).getParent().toString();
       if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
```

Comment:
The problem occurs in S3 because S3 doesn't have a concept of file path, it only has key/value, so there needs to be logic to extract object key from file path in my reference above, and the key extraction logic depends on the scheme. 

HDFS only needs a file path, so it shouldn't have a problem.